server/master/src/main/java/org/apache/accumulo/master/tableOps/CreateInitialSplits.java

```
@@ -0,0 +1,179 @@
+      writeLastRowToMetadataTable(splitEntry, bw);
+    }
+
+    return new FinishCreateTable(tableInfo);
+  }
+
+  private void writeLastRowToMetadataTable(SplitEntry splitEntry, BatchWriter bw)
+      throws MutationsRejectedException {
+    Text lastRow = new Text(tableInfo.tableId.toString() + "<");
+    Mutation mut = new Mutation(lastRow);
+    mut.put(new Text("srv"), new Text("dir"), splitEntry.dirValue);
+    mut.put(new Text("srv"), new Text("lock"), splitEntry.lockValue);
+    mut.put(new Text("srv"), new Text("time"), splitEntry.timeValue);
+    mut.put(new Text("~tab"), new Text("~pr"),
+        KeyExtent.encodePrevEndRow(new Text(splitEntry.lastSplit)));
+    bw.addMutation(mut);
+  }
+
+  private void writeSplitsToMetadataTable(Master environment, String splitFile,
+      SplitEntry splitEntry, BatchWriter bw) throws IOException {
+    String tableId = tableInfo.tableId.toString();
+    try (FSDataInputStream splitStream = environment.getInputStream(splitFile)) {
+      try (BufferedReader br = new BufferedReader(new InputStreamReader(splitStream))) {
+        String split;
+        while ((split = br.readLine()) != null) {
+          try {
+            Mutation mut = new Mutation(tableId + ";" + split);
+            mut.put(new Text("srv"), new Text("dir"), splitEntry.dirValue);
+            mut.put(new Text("srv"), new Text("lock"), splitEntry.lockValue);
+            mut.put(new Text("srv"), new Text("time"), splitEntry.timeValue);
+            if (splitEntry.first) {
+              mut.put(new Text("~tab"), new Text("~pr"), splitEntry.firstRow);
+              splitEntry.first = false;
+            } else {
+              mut.put(new Text("~tab"), new Text("~pr"),
+                  KeyExtent.encodePrevEndRow(new Text(splitEntry.lastSplit)));
+            }
+            bw.addMutation(mut);
+            splitEntry.lastSplit = split;
+          } catch (Exception e) {
+            e.printStackTrace();
```

Should be a logger to log here.

---

Discussed with @keith-turner and decided that initially will let the method throw an exception if it occurs. The FATE server should log the event. After initial release will examine behaviour of DFS and determine if we need to add some type of retry in the event of failure.